src/main/java/de/tum/in/www1/exerciseapp/service/StatisticService.java

```diff
@@ -120,44 +127,79 @@ public void updateStatisticsAndResults(QuizExercise quizExercise){
                 quizSubmissionRepository.save(quizSubmission);
 
                 // find latest rated Result
-                if (result.getCompletionDate().isBefore(quizExercise.getDueDate())
-                    && (latestRatedResult == null || latestRatedResult.getCompletionDate().isBefore(result.getCompletionDate()))) {
+                if (result.isRated() && (latestRatedResult == null || latestRatedResult.getCompletionDate().isBefore(result.getCompletionDate()))) {
                     latestRatedResult = result;
                 }
                 // find latest unrated Result
-                if (result.getCompletionDate().isAfter(quizExercise.getDueDate())
-                    && (latestUnratedResult == null || latestUnratedResult.getCompletionDate().isBefore(result.getCompletionDate()))) {
+                if (!result.isRated() && (latestUnratedResult == null || latestUnratedResult.getCompletionDate().isBefore(result.getCompletionDate()))) {
                     latestUnratedResult = result;
                 }
             }
             // update statistics with latest rated und unrated Result
-            this.addResultToAllStatistics(quizExercise, latestRatedResult, true);
-            this.addResultToAllStatistics(quizExercise, latestUnratedResult, false);
+            this.addResultToAllStatistics(quizExercise, latestRatedResult);
+            this.addResultToAllStatistics(quizExercise, latestUnratedResult);
 
         }
         //save changed Statistics
         quizPointStatisticRepository.save(quizExercise.getQuizPointStatistic());
         for (Question question : quizExercise.getQuestions()) {
-            questionStatisticRepository.save(question.getQuestionStatistic());
+            if (question.getQuestionStatistic() != null) {
+                questionStatisticRepository.save(question.getQuestionStatistic());
+            }
+        }
+    }
+
+    public boolean addAndRemoveResultToStatistics(Participation participation, Result newResult, Result oldResult) {
```

Comment:
please call this method updateStatistics and add JavaDoc. I don't really get what this method is doing.

Reply:
@MoritzIssig will do this.